src/Acuminator/Acuminator.Analyzers/StaticAnalysis/PublicClassXmlComment/PublicClassXMLCommentAnalyzer.cs

```diff
@@ -0,0 +1,172 @@
+			: this(null)
+		{
+		}
+
+        internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+        {
+            compilationStartContext.RegisterSyntaxNodeAction(AnalyzeClassDeclaration, SyntaxKind.ClassDeclaration);
+        }
+
+        private void AnalyzeClassDeclaration(SyntaxNodeAnalysisContext syntaxContext)
+        {
+            syntaxContext.CancellationToken.ThrowIfCancellationRequested();
+
+            if (!(syntaxContext.Node is ClassDeclarationSyntax classDeclarationSyntax))
+            {
+                return;
+            }
+
+            var isPublicClass = classDeclarationSyntax.Modifiers.Any(SyntaxKind.PublicKeyword);
+            if (!isPublicClass)
+            {
+                return;
+            }
+
+            var isAcumaticaClass = IsInAcumaticaRootNamespace(classDeclarationSyntax, syntaxContext.SemanticModel, syntaxContext.CancellationToken);
+            if (!isAcumaticaClass)
+            {
+                return;
+            }
+
+            CheckXmlComment(syntaxContext, classDeclarationSyntax);
+        }
+
+        private bool IsInAcumaticaRootNamespace(ClassDeclarationSyntax classDeclarationSyntax, SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var classDeclarationSymbol = semanticModel.GetDeclaredSymbol(classDeclarationSyntax, cancellationToken);
+            if (classDeclarationSymbol == null)
+            {
+                return false;
+            }
+
+			var classRootNamespace = classDeclarationSymbol
+				.GetContainingNamespaces()
+				.Where(n => !string.IsNullOrEmpty(n.Name))
+				.Last();
+            if (!NamespaceNames.AcumaticaRootNamespace.Equals(classRootNamespace.Name, StringComparison.Ordinal))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        private void CheckXmlComment(SyntaxNodeAnalysisContext syntaxContext, ClassDeclarationSyntax classDeclarationSyntax)
+        {
+            syntaxContext.CancellationToken.ThrowIfCancellationRequested();
+
+            if (!classDeclarationSyntax.HasStructuredTrivia)
+            {
+                ReportDiagnostic(syntaxContext, classDeclarationSyntax, FixOption.NoXmlComment);
+				return;
+            }
+
+            var xmlComment = classDeclarationSyntax
+                .GetLeadingTrivia()
+                .Select(t => t.GetStructure())
+                .OfType<DocumentationCommentTriviaSyntax>()
+				.FirstOrDefault();
+            if (xmlComment == null)
+            {
+                ReportDiagnostic(syntaxContext, classDeclarationSyntax, FixOption.NoXmlComment);
+				return;
+            }
+
+			var excludeTag = xmlComment
+				.ChildNodes()
+				.OfType<XmlEmptyElementSyntax>()
+				.Where(s => XmlCommentExcludeTag.Equals(s?.Name?.ToString(), StringComparison.Ordinal))
```

Comment:
A little nitpicking so I won't insist on it but Where and FirstOrDefault can be combined in one:
```csharp
FirstOrDefault(s => XmlCommentExcludeTag.Equals(s?.Name?.ToString(), StringComparison.Ordinal))
```

Reply:
It was done on purpose. I'm prefer such chain of Linq methods in which every method performs single operation. In this particular case it clearly indicates where we filter results (`Where`) and where we control what element of filtered results we want to use (`FirstOrDefault`).

Reply:
Ok, I won't insist on it